Code/src/main/java/nl/utwente/group10/ui/components/FunctionBlock.java

```diff
@@ -23,17 +28,20 @@
 	private String[] arguments;
 	/** The name of this Function.**/
 	private String functionName;
-	
+	/** intstance to create Events for this FunctionBlock. **/
+	private static CustomGesture cg;
+		
 	/**
 	 * Method that creates a newInstance of this class along with it's visual representation
 	 * @param the number of arguments this FunctionBlock can hold
 	 * @return a new instance of this class
 	 * @throws IOException
 	 */
-	public static FunctionBlock newInstance(int numberOfArguments) throws IOException {
+	public static FunctionBlock newInstance(int numberOfArguments, CustomUIPane pane) throws IOException {
```

Comment:
Yes, that would be the 'walking the tree upwards' approach I think. All of the four approaches are reasonable, really, it's just a matter of preference.